### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Meklo**
Went from : 
```
@gridsuite/commons-ui@0.51.0 /home/marcelinhug/Projects/workspace/frontend/commons-ui
├── @babel/helper-builder-react-jsx@7.15.4
├── @babel/preset-env@7.23.8
├── @babel/preset-react@7.23.3
├── @babel/preset-typescript@7.23.3
├── @emotion/react@11.10.6
├── @emotion/styled@11.10.6
├── @hookform/resolvers@3.2.0
├── @mui/icons-material@5.11.16
├── @mui/lab@5.0.0-alpha.127
├── @mui/material@5.12.1
├── @mui/styles@5.12.0
├── @react-hook/window-size@3.1.1
├── @types/react-dom@18.2.19
├── @types/react@18.2.60
├── @typescript-eslint/eslint-plugin@5.62.0
├── @typescript-eslint/parser@5.62.0
├── @vitejs/plugin-react@4.2.1
├── autosuggest-highlight@3.3.4
├── babel-eslint@10.1.0
├── bufferutil@4.0.7
├── clsx@1.2.1
├── eslint-config-prettier@8.8.0
├── eslint-config-react-app@6.0.0
├── eslint-plugin-flowtype@5.10.0
├── eslint-plugin-import@2.27.5
├── eslint-plugin-jsx-a11y@6.7.1
├── eslint-plugin-prettier@4.2.1
├── eslint-plugin-react-hooks@4.6.0
├── eslint-plugin-react@7.32.2
├── eslint@7.32.0
├── glob@10.3.10
├── jest-environment-jsdom@29.7.0
├── jest-svg-transformer@1.0.0
├── jest@29.7.0
├── jwt-decode@3.1.2
├── memoize-one@6.0.0
├── notistack@3.0.1
├── oidc-client@1.11.5
├── prettier@2.8.7
├── prop-types@15.8.1
├── react-csv-downloader@2.9.1
├── react-dom@18.2.0
├── react-hook-form@7.45.4
├── react-intl@6.4.1
├── react-request-fullscreen@1.1.2
├── react-resizable@3.0.5
├── react-router-dom@6.10.0
├── react-virtualized@9.22.5
├── react@18.2.0
├── ts-node@10.9.2
├── type-fest@2.19.0
├── typescript@5.1.6
├── utf-8-validate@5.0.10
├── uuid@3.4.0
├── vite-plugin-dts@2.3.0
├── vite-plugin-eslint@1.8.1
├── vite-plugin-lib-inject-css@2.0.0
├── vite-plugin-svgr@4.2.0
├── vite@5.1.3
└── yup@1.2.0
```
to 
```
@gridsuite/commons-ui@0.52.0 /home/marcelinhug/Projects/workspace/frontend/commons-ui
├── @babel/helper-builder-react-jsx@7.22.10
├── @babel/preset-env@7.24.3
├── @babel/preset-react@7.24.1
├── @babel/preset-typescript@7.24.1
├── @emotion/react@11.11.4
├── @emotion/styled@11.11.5
├── @hookform/resolvers@3.3.4
├── @mui/icons-material@5.15.14
├── @mui/lab@5.0.0-alpha.169
├── @mui/material@5.15.14
├── @mui/styles@5.15.14
├── @mui/x-tree-view@6.17.0
├── @react-hook/window-size@3.1.1
├── @types/react-dom@18.2.23
├── @types/react@18.2.73
├── @typescript-eslint/eslint-plugin@6.21.0
├── @typescript-eslint/parser@6.21.0
├── @vitejs/plugin-react@4.2.1
├── autosuggest-highlight@3.3.4
├── babel-eslint@10.1.0
├── bufferutil@4.0.8
├── clsx@2.1.0
├── eslint-config-prettier@9.1.0
├── eslint-config-react-app@6.0.0
├── eslint-plugin-flowtype@8.0.3
├── eslint-plugin-import@2.29.1
├── eslint-plugin-jsx-a11y@6.8.0
├── eslint-plugin-prettier@4.2.1
├── eslint-plugin-react-hooks@4.6.0
├── eslint-plugin-react@7.34.1
├── eslint@7.32.0
├── glob@10.3.12
├── jest-environment-jsdom@29.7.0
├── jest-svg-transformer@1.0.0
├── jest@29.7.0
├── jwt-decode@4.0.0
├── license-checker@25.0.1
├── memoize-one@6.0.0
├── notistack@3.0.1
├── oidc-client@1.11.5
├── prettier@2.8.8
├── prop-types@15.8.1
├── react-csv-downloader@3.1.0
├── react-dom@18.2.0
├── react-hook-form@7.51.2
├── react-intl@6.6.4
├── react-request-fullscreen@1.1.2
├── react-resizable@3.0.5
├── react-router-dom@6.22.3
├── react-virtualized@9.22.5
├── react@18.2.0
├── ts-node@10.9.2
├── type-fest@4.14.0
├── typescript@5.1.6
├── utf-8-validate@6.0.3
├── uuid@9.0.1
├── vite-plugin-dts@3.8.1
├── vite-plugin-eslint@1.8.1
├── vite-plugin-lib-inject-css@2.0.1
├── vite-plugin-svgr@4.2.0
├── vite@5.2.7
└── yup@1.4.0
```

### CODE REVIEW

**Comment by flomillot** [comment]
When you build there is lot of warnings now :

```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.2.0

YOUR TYPESCRIPT VERSION: 5.4.3

Please only submit bug reports when using the officially supported version.
```


```
One of your dependencies, babel-preset-react-app, is importing the
"@babel/plugin-proposal-private-property-in-object" package without
declaring it in its dependencies. This is currently working because
"@babel/plugin-proposal-private-property-in-object" is already in your
node_modules folder for unrelated reasons, but it may break at any time.

babel-preset-react-app is part of the create-react-app project, which
is not maintianed anymore. It is thus unlikely that this bug will
ever be fixed. Add "@babel/plugin-proposal-private-property-in-object" to
your devDependencies to work around this error. This will make this message
go away.
```
  
That's why I used some specific versions, to remove all these warnings.
You can ask a TL but IMO we should keep the previous versions until we totally clean all the eslint deps in an other TS.

File: package.json [hunk 1]
```diff
@@ -27,83 +27,84 @@
...
-        "@hookform/resolvers": "^3.0.0",
         "notistack": "^3.0.0",
-        "react-router-dom": "^6.0.0",
         "react": "^18.0.0",
         "react-dom": "^18.0.0",
-        "react-intl": "^6.0.0",
         "react-hook-form": "^7.41.0",
+        "react-intl": "^6.0.0",
+        "react-router-dom": "^6.0.0",
         "yup": "^1.0.0"
```
Comment (Tristan-WorkGH) [inline package.json:54]: `peerDependencies` aren't in sync with `dependencies` and `devDependencies` versions

File: vite.config.mts [hunk 2]
```diff
@@ -72,7 +72,7 @@ export default defineConfig((config) => ({
 
 // Workaround for react-virtualized with vite
 // https://github.com/bvaughn/react-virtualized/issues/1632#issuecomment-1483966063
-function reactVirtualized(): PluginOption {
+function reactVirtualized() {
```
Comment (Tristan-WorkGH) [inline vite.config.mts:75]: Why did you remove the type? TypeScript didn't infer correctly in IDE inside the function.

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]
`package-lock.json`:
```shell
$ rm -rf node_modules && npm ci && npm outdated --depth=0
Package           Current   Wanted   Latest  Location                       Depended by
@mui/x-tree-view   6.17.0   6.17.0    7.0.0  node_modules/@mui/x-tree-view  commons-ui
@types/react      18.2.71  18.2.73  18.2.73  node_modules/@types/react      commons-ui
react-hook-form    7.51.1   7.51.2   7.51.2  node_modules/react-hook-form   commons-ui
react-intl          6.6.3    6.6.4    6.6.4  node_modules/react-intl        commons-ui
vite-plugin-dts     3.3.0    3.3.0    3.8.0  node_modules/vite-plugin-dts   commons-ui
```

`package.json`:
```shell
$ rm -rf node_modules && rm package-lock.json && sed -i 's/\^//g' package.json && npm install && npm outdated --depth=0
Package           Current   Wanted   Latest  Location                       Depended by
@mui/x-tree-view   6.17.0   6.17.0    7.0.0  node_modules/@mui/x-tree-view  commons-ui
@types/react      18.2.71  18.2.71  18.2.73  node_modules/@types/react      commons-ui
react-hook-form    7.51.1   7.51.1   7.51.2  node_modules/react-hook-form   commons-ui
react-intl          6.6.3    6.6.3    6.6.4  node_modules/react-intl        commons-ui
vite-plugin-dts     3.3.0    3.3.0    3.8.0  node_modules/vite-plugin-dts   commons-ui
```
⚠️ `package-lock` is head of `package` on versions.  
We have `vite-plugin-dts`, `@types/react`, `react-hook-form` and `react-intl` to update and it's all good for me 👍🏻

With below comment of course 😉

**Comment by Tristan-WorkGH** [comment]
Note for PR:  
migration to `@mui/x-tree-view@7.x` will be for another PR as changes are to important.

File: vite.config.mts (hunk 2 above)
Comment (Tristan-WorkGH) [reply vite.config.mts:75]: ah I understand
```
/home/chuinetri/Projects/__git/_gridsuite/commons-ui/vite.config.mts
  10:15  warning  'PluginOption' is defined but never used  no-unused-vars
```

It's a error of eslint, the type is used 😕  
I add a ignore on my branch

File: package.json (hunk 1 above)
Comment (Meklo) [reply package.json]: Done

**Comment by Tristan-WorkGH** [comment]
ℹ️ Note for PR:  
`eslint`, `eslint-config-react-app`, `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` are to be updated in another PR because of incompatibilities between them, and some are archived & obsolete.

**Review by Tristan-WorkGH — APPROVED** [review]
`package-lock.json`:
```shell
$ rm -rf node_modules && npm ci && npm outdated --depth=0
Package                           Current   Wanted   Latest  Location                                       Depended by
@mui/x-tree-view                   6.17.0   6.17.0    7.1.0  node_modules/@mui/x-tree-view                  commons-ui
@types/react-dom                  18.2.22  18.2.23  18.2.23  node_modules/@types/react-dom                  commons-ui
@typescript-eslint/eslint-plugin   6.21.0   6.21.0    7.4.0  node_modules/@typescript-eslint/eslint-plugin  commons-ui
@typescript-eslint/parser          6.21.0   6.21.0    7.4.0  node_modules/@typescript-eslint/parser         commons-ui
eslint                             7.32.0   7.32.0   8.57.0  node_modules/eslint                            commons-ui
eslint-config-react-app             6.0.0    6.0.0    7.0.1  node_modules/eslint-config-react-app           commons-ui
glob                              10.3.10  10.3.12  10.3.12  node_modules/glob                              commons-ui
typescript                          5.1.6    5.1.6    5.4.3  node_modules/typescript                        commons-ui
```

`package.json`:
```shell
$ rm -rf node_modules && rm package-lock.json && sed -i 's/\^//g' package.json && npm install && npm outdated --depth=0
Package                           Current   Wanted   Latest  Location                                       Depended by
@mui/x-tree-view                   6.17.0   6.17.0    7.1.0  node_modules/@mui/x-tree-view                  commons-ui
@types/react-dom                  18.2.22  18.2.22  18.2.23  node_modules/@types/react-dom                  commons-ui
@typescript-eslint/eslint-plugin   6.21.0   6.21.0    7.4.0  node_modules/@typescript-eslint/eslint-plugin  commons-ui
@typescript-eslint/parser          6.21.0   6.21.0    7.4.0  node_modules/@typescript-eslint/parser         commons-ui
eslint                             7.32.0   7.32.0   8.57.0  node_modules/eslint                            commons-ui
eslint-config-react-app             6.0.0    6.0.0    7.0.1  node_modules/eslint-config-react-app           commons-ui
glob                              10.3.10  10.3.10  10.3.12  node_modules/glob                              commons-ui
typescript                          5.1.6    5.1.6    5.4.3  node_modules/typescript                        commons-ui
```

Good for me, even if there _now_ `@types/react-dom` and `glob` with a patch available.

File: vite.config.mts
Comment (Tristan-WorkGH) [inline vite.config.mts:1]: I just tested with last modifications, you can restore the type, it work now.

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: vite.config.mts
Comment (Meklo) [reply vite.config.mts:1]: Indeed, done

**Review by Tristan-WorkGH — APPROVED** [review]

**Comment by jonenst** [comment]
forgot the one in 
```
diff --git a/src/components/ReportViewer/report-item.jsx b/src/components/ReportViewer/report-item.jsx
index e9ee6d3..7afed79 100644
--- a/src/components/ReportViewer/report-item.jsx
+++ b/src/components/ReportViewer/report-item.jsx
@@ -9,7 +9,7 @@ import { useContext } from 'react';
 import { Box, Typography } from '@mui/material';
 import { alpha, styled } from '@mui/system';
 import PropTypes from 'prop-types';
-import { TreeItem } from '@mui/lab';
+import { TreeItem } from '@mui/x-tree-view';
 import { Label } from '@mui/icons-material';
 import ReportTreeViewContext from './report-tree-view-context';
 ```
 
 reportviewer has been blank in the demo ever since
